### DISCUSSION TITLE
Opacity Slider

### DISCUSSION THREAD

**Discussion by pd-allen** [banesullivan/localtileserver#78] [Q&A] [answered]
The local tile server implementation is very useful. Is there a way to embed an Opacity slider into an html file for inserted tiles? I am using your code to overlay World War One Trench maps over current maps and it is desirable to vary the opacity to look at features on the current map.

Thanks
Paul Allen

**Comment by banesullivan** [comment]
There definitely is an easy way to do this!

But first, how are you displaying the tiles? Can you share code? Are you using CesiumJS, leaflet, openlayers, or what?

**Reply by pd-allen** [reply]
I am using folium.  Here is the code, I want to tie the slide to opval.

```py
import folium
from folium import plugins
import sys
import os.path
from folium import IFrame
from folium.plugins import TimestampedGeoJson,Geocoder,MeasureControl,MiniMap,MousePosition,MarkerCluster
from ipywidgets import interact
from localtileserver import get_folium_tile_layer
from localtileserver import TileClient
merc = os.path.join("c:\python\data", "Map28NW4.tif")
merclen=int(len(merc)-4)
mercname=merc[15:merclen]
print(mercname)
tile_client = TileClient(merc)    
opval=0.8
t = get_folium_tile_layer(tile_client,opacity=opval,overlay='true',name='Map28NW4')
myMap = folium.Map(location = tile_client.center(),zoom_start = 9,height='96%',control_scale = True,overlay='false')
myMap.add_child(t)
print(tile_client.center(),opval)
display(myMap)
myMap.save('ypres.html')
```

**Reply by banesullivan** [reply]
We could use ipywidgets to add a slider widget in the jupyter notebook, but since your goal is to create a standalone app, I think streamlit with what's posted below is the best route for you

**Reply by pd-allen** [reply]
Thanks to you both. I've started to look at streamlit and think I can get what I need.

**Comment by giswqs** [comment]
Try [leafmap](https://github.com/giswqs/leafmap/). You can change layer opacity without coding. 
https://leafmap.org/notebooks/32_local_tile

![](https://i.imgur.com/3ZltNUc.gif)

**Reply by pd-allen** [reply]
The leafmap module is very comprehensive. I appreciate the documentation and large number of tutorials. Is there a way to enable the toolbox in streamlit? I implemented an opacity slider for my map tile in streamlit following your example in the geemap module. It takes several seconds to reload the map with the new level, and is not a smooth as the toolbox slider.

**Reply by giswqs** [reply]
The toolbox is for ipyleaflet only. Unfortunately, streamlit does not support ipyleaflet.

**Reply by pd-allen** [reply]
Thanks for your help.

**Comment by giswqs** [comment, marked as answer]
![Peek 2022-04-20 17-43](https://user-images.githubusercontent.com/5016453/164328658-aed82da1-d8e4-42ec-8644-dbf5312ebbb1.gif)

**Reply by pd-allen** [reply]
That is very slick, thanks for the quick response. Is there any way to have the sliders available in an html file?  Some of the folks looking at the maps will not have notebook available.

**Reply by giswqs** [reply]
Unfortunately, the answer is no. ipywidgets won't work in HTML. One workaround is to use streamlit 

https://github.com/giswqs/streamlit-template

**Reply by pd-allen** [reply]
Thanks, I'll give it a try.

**Reply by banesullivan** [reply]
@pd-allen, I think streamlit is the right choice for you. localtileserver requires a live python thread (which can't ship in a standalone HTML file). Streamlit will give you the ability to have a web application that serves the tiles of your maps into your map viewer
